internal/engine/evaluator.go

```diff
@@ -158,6 +153,21 @@ func (rpe *resourcePolicyEvaluator) Evaluate(ctx context.Context, tctx tracer.Co
 			dctx.Activated()
 		}
 
+		if len(effectiveDerivedRoles) > 0 {
+			// populate `P.derivedRoles` for use in rules
+			for derivedRole := range effectiveDerivedRoles {
+				request.Principal.DerivedRoles = append(request.Principal.DerivedRoles, derivedRole)
+			}
+			sort.Strings(request.Principal.DerivedRoles)
```

Comment:
Could we save a few CPU cycles by not sorting this? Most of the time it's going to be unused and when it's do used, it's most likely to be in an expression like `x in P.derivedRoles`.

Reply:
I think it's nicer if it's stable (it could be useful in outputs too). However, I think we can do it at compile time instead: https://github.com/cerbos/cerbos/pull/1778#issuecomment-1719097314